### PR TITLE
Add tests to check that device.log was created in all cases

### PR DESCRIPTION
**Pull request by jonathanmos**
## Description
Adds test coverage for the creation of the device.log. Fixes issue #3469

### CODE REVIEW

**Review by noomorph — APPROVED** [review]
Let's wait for CI to pass again, and LGTM.

File: detox/test/e2e/21.artifacts.test.js
Comment (asafkorem) [inline detox/test/e2e/21.artifacts.test.js:129]: @noomorph why `beforeAll`?

**Pull request merged**
